Presentation/MasterDetailSelection.swift

```diff
@@ -21,19 +21,24 @@ public final class MasterDetailSelection<Elements: BidirectionalCollection>: Sig
     private let bag = DisposeBag()
     fileprivate let keepSelection: KeepSelection<Elements>
     private var isSelecting = false
-    fileprivate let isCollapsed: ReadSignal<Bool>
+    fileprivate let isCollapsed: ReadSignal<Bool?>
+
+    @available(*, deprecated, message: "use `init(elements:isSame:needsUpdate:collapsedState:)` instead")
+    public convenience init(elements: ReadSignal<Elements>, isSame: @escaping (Element, Element) -> Bool, needsUpdate: @escaping (Element, Element) -> Bool = { _, _ in false }, isCollapsed: ReadSignal<Bool>) {
+        self.init(elements: elements, isSame: isSame, needsUpdate: needsUpdate, collapsedState: isCollapsed.map { value -> Bool? in return value })
+    }
 
     /// Creates a new instance using changes in `elements` and `isCollapsed` to maintain the selected detail index (provided signal).
     /// - Parameters:
     ///   - isSame: Is it the same row (same identity)
     ///   - needsUpdate: For the same row, does the row have updates that requires presenting new details (refresh details)
     ///   - isCollapsed: Whether or not details are displayed.
-    public init(elements: ReadSignal<Elements>, isSame: @escaping (Element, Element) -> Bool, needsUpdate: @escaping (Element, Element) -> Bool = { _, _ in false }, isCollapsed: ReadSignal<Bool>) {
+    public init(elements: ReadSignal<Elements>, isSame: @escaping (Element, Element) -> Bool, needsUpdate: @escaping (Element, Element) -> Bool = { _, _ in false }, collapsedState: ReadSignal<Bool?>) {
```

Comment:
I wonder if we can keep the same parameter name `isCollapsed`, but just change the type and let both the deprecated and the new init live side by side only differ by the signal name?

Reply:
Seems like it's working 👍